controllers/tasks.js

```diff
@@ -321,12 +321,28 @@ const updateTaskStatus = async (req, res, next) => {
     if (task.taskData.status === TASK_STATUS.VERIFIED || req.body.status === TASK_STATUS.MERGED)
       return res.boom.forbidden("Status cannot be updated. Please contact admin.");
 
+    if (userStatusFlag) {
+      if (task.taskData.status === TASK_STATUS.DONE && req.body.percentCompleted < 100) {
+        if (req.body.status === TASK_STATUS.DONE || !req.body.status) {
+          return res.boom.badRequest("Task percentCompleted can't updated as status is DONE");
+        }
+      }
+
+      if (
+        (req.body.status === TASK_STATUS.DONE || req.body.status === TASK_STATUS.VERIFIED) &&
+        task.taskData.percentCompleted !== 100
+      ) {
+        if (req.body.percentCompleted !== 100) {
+          return res.boom.badRequest("Status cannot be updated. Task is not done yet");
+        }
```

Comment:
here also why is the precentageCompeled !== 100 being checked twice?

Reply:
Reason found from vinayak-g who written this code
<img width="675" alt="image" src="https://github.com/Real-Dev-Squad/website-backend/assets/107163260/3537ff25-342e-40a8-9718-3bc8a94bc04f">

https://github.com/Real-Dev-Squad/website-backend/pull/725#discussion_r998594473